### ISSUE TITLE
Replace optimist or otherwise avoid its vulnerability

### ISSUE THREAD

**Issue by brettz9** [cloudhead/node-static#218]
The dependency `optimist` depends on a vulnerable version of `minimist` (see https://npmjs.com/advisories/1179).

Despite having an [issue](https://github.com/substack/node-optimist/issues/152) and [PR](https://github.com/substack/node-optimist/pull/150) filed with `optimist`, since the last commit on optimist was 6 years ago, and since the package is [deprecated](https://github.com/substack/node-optimist#deprecation-notice) (and it has also been [awaiting a fix](https://github.com/substack/node-optimist/pull/149) for a proper license specifier, making automated license audits problematic in the interim), it'd be really nice if you could drop the `optimist` dependency, perhaps using one of its suggested replacements (yargs, nomnom, or using `minimist` directly, though if the latter, I'd hope pegging against the maintained major bump).

(I personally like `command-line-args`, as one can use it with `command-line-usage` or my `command-line-publish`/`command-line-basics` tools to get documentation (both at the command line and as SVG, allowing embedding in a README) from a simple declarative schema.)

Thanks!

**Comment by debrice** [comment]
@brettz9 You can enforce the right version of `minimist` by using the `resolutions` feature in your `package.json` file. I've successfully removed the vulnerability [through this commit](https://github.com/briga-software/dummy-backend/commit/1292de0879774e62a5ed3853de336bec4d1e67fb#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R28-R30)

**Comment by brettz9** [comment]
@debrice: Yes, thank you, but not every project is using Yarn.

**Comment by Prajwal-Koirala** [comment]
+1

**Comment by dscham** [comment]
Bump

**Comment by Jahmilli** [comment]
I might be mistaken but I found that installing [npm-force-resolutions](https://www.npmjs.com/package/npm-force-resolutions) and adding the following into the package.json
```
  "resolutions": {
    "minimist": "^1.2.5",
  },
```
as well as the following script into my package.json: `"preinstall": "npx npm-force-resolutions"`
By then running `npm run preinstall` and then `npm i` it seemed to remove the audit issue around using minimist.

**Comment by dscham** [comment]
Looks like it's dead, last commit on master in 2018? Is there still somebody to merge? Then I maybe make a fork with a fix.

**Comment by rfossella** [comment]
Is  installing **npm-force-resolutions** as a dev dependency still the only (recommended) solution to the optimist/minimist issue?

> npm audit
```
Low             Prototype Pollution
  Package         minimist
  Patched in      >=0.2.1 <1.0.0 || >=1.2.3
  Dependency of   optimist
  Path            optimist > minimist
  More info       https://npmjs.com/advisories/1179
```

**Comment by brettz9** [comment]
Closed by #227.
